variants/STM32F4xx/F405RGT_F415RGT/variant_MICROMOD_F405.h

```diff
@@ -0,0 +1,246 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2017, STMicroelectronics
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of STMicroelectronics nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************
+ */
+#pragma once
+
+#define USBD_VID  0x1B4F
+#define USBD_PID  0x0029
+
+/*----------------------------------------------------------------------------
+ *        Pins
+ *----------------------------------------------------------------------------*/
+#define PC0  0  //D0
+#define PC1  1  //D1
+
+#define PC6  2  //PWM0
+#define PC7  3  //PWM1
+
+#define PA2  4  //TX1
+#define PA3  5  //RX1
+
+#define PD2  6  //G0
+#define PA8  7  //G1
+#define PA0  8  //G2
+#define PC8  9  //G3
+#define PC9  10  //G4
+#define PC13 11  //G5
+#define PC2  12  //G6
+#define PB13 13  //G10 - HOST_VBUS
+#define PB12 14  //G11 - HOST_ID
+
+#define PB4  15  //AUD_OUT
+#define PB5  16  //AUD_IN
+#define PA4  17  //AUD_LRCLK
+#define PB3  18  //AUD_BCLK  
+
+#define PB9  19  //CAN_TX
+#define PB8  20  //CAN_RX
+
+#define PB10 21  //SCL
+#define PB11 22  //SDA
+#define PB1  23  //INT
+
+#define PB6  24  //SCL1
+#define PB7  25  //SDA1
+
+#define PA5  26  //SCK
+#define PA7  27  //COPI
+#define PA6  28  //CIPO
+#define PC4  29  //CS
+
+#define PB14 30  //HOST_D-
+#define PB15 31  //HOST_D+
+
+#define PA15 32  //STAT LED
+
+#define PC5  33  //A0
+#define PB0  34  //A1
```

Comment:
For analog pin then define like this, like this PYn alias can be used with analog API:
```suggestion
#define PC5  PIN_A0
#define PB0  PIN_A1
```
Same for other analog pins defined on the analog array